node/src/eth/api.rs

```diff
@@ -0,0 +1,704 @@
+use crate::{
+    eth::{
+        backend,
+        error::{BlockchainError, FeeHistoryError, Result, ToRpcResponseResult},
+        fees::{FeeDetails, FeeHistory, FeeHistoryCache},
+        pool::{
+            transactions::{to_marker, PoolTransaction},
+            Pool,
+        },
+        sign::Signer,
+    },
+    revm::TransactOut,
+};
+use ethers::{
+    abi::ethereum_types::H64,
+    types::{
+        Address, Block, BlockNumber, Bytes, Log, Transaction, TransactionReceipt, TxHash, H256,
+        U256, U64,
+    },
+    utils::rlp,
+};
+use foundry_node_core::{
+    eth::{
+        call::CallRequest,
+        filter::Filter,
+        transaction::{
+            EthTransactionRequest, LegacyTransaction, PendingTransaction, TypedTransaction,
+            TypedTransactionRequest,
+        },
+        EthRequest,
+    },
+    response::ResponseResult,
+    types::{Index, Work},
+};
+use std::sync::Arc;
+
+/// The entry point for executing eth api RPC call - The Eth RPC interface.
+///
+/// This type is cheap to clone and can be used concurrently
+#[derive(Clone)]
+pub struct EthApi {
+    /// The transaction pool
+    pool: Arc<Pool>,
+    /// Holds all blockchain related data
+    /// In-Memory only for now
+    backend: Arc<backend::mem::Backend>,
+    /// Whether this node is mining
+    is_authority: bool,
+    /// available signers
+    signers: Arc<Vec<Box<dyn Signer>>>,
```

Comment:
wait I thought `Signer` wasn't object safe...oops? not for this PR but probably means we should refactor these `match` arms to use a trait object Signer instead of concrete https://github.com/gakonst/foundry/blob/master/cli/src/cast.rs#L266

Reply:
this is a new signer trait,
and the `DevSigner` implements that by wrapping `ethers::Wallet`